docs/blog/posts/cleanrl-v1.md

```diff
@@ -0,0 +1,268 @@
+
+```python
+import random
+from typing import Callable
+
+import gym
+import numpy as np
+import torch
+
+
+def evaluate(
+    model_path: str,
+    make_env: Callable,
+    env_id: str,
+    eval_episodes: int,
+    run_name: str,
+    Model: torch.nn.Module,
+    device: torch.device,
+    epsilon: float = 0.05,
+    capture_video: bool = True,
+):
+    envs = gym.vector.SyncVectorEnv([make_env(env_id, 0, 0, capture_video, run_name)])
+    model = Model(envs).to(device)
+    model.load_state_dict(torch.load(model_path))
+    model.eval()
+
+    obs = envs.reset()
+    episodic_returns = []
+    while len(episodic_returns) < eval_episodes:
+        if random.random() < epsilon:
+            actions = np.array([envs.single_action_space.sample() for _ in range(envs.num_envs)])
+        else:
+            q_values = model(torch.Tensor(obs).to(device))
+            actions = torch.argmax(q_values, dim=1).cpu().numpy()
+        next_obs, _, _, infos = envs.step(actions)
+        for info in infos:
+            if "episode" in info.keys():
+                print(f"eval_episode={len(episodic_returns)}, episodic_return={info['episode']['r']}")
+                episodic_returns += [info["episode"]["r"]]
+        obs = next_obs
+
+    return episodic_returns
+
+
+if __name__ == "__main__":
+    from huggingface_hub import hf_hub_download
+
+    from cleanrl.dqn import QNetwork, make_env
+
+    model_path = hf_hub_download(repo_id="cleanrl/CartPole-v1-dqn-seed1", filename="q_network.pth")
+```
+
+
+<img src="/static/blog/cleanrl-v1/hf.png">
+
+### RLops
+
+How do we know the effect of a new feature / bug fix? DRL is brittle and has a series of reproducibility issues — even bug fixes sometimes could introduce performance regression (e.g., see [how a bug fix of contact force in MuJoCo results in worse performance for PPO](https://github.com/openai/gym/pull/2762#discussion_r853488897)). Therefore, it is essential to understand how the proposed changes impact the performance of the algorithms. 
+
+We built a prototype tool that allows us to compare the performance of the library at different versions of the tracked experiment. With this tool, we can confidently merge new features / bug fixes without worrying about introducing catastrophic regression. The users can run commands such as:
+
+```bash
+python -m cleanrl_utils.rlops --exp-name ddpg_continuous_action \
+    --wandb-project-name cleanrl \
+    --wandb-entity openrlbenchmark \
+    --tags 'pr-299' 'rlops-pilot' \
+    --env-ids HalfCheetah-v2 Walker2d-v2 Hopper-v2 InvertedPendulum-v2 Humanoid-v2 Pusher-v2 \
+    --output-filename compare.png \
+    --scan-history \
+    --metric-last-n-average-window 100 \
+    --report
+```
+
+which generates the following image
+
+<img max-width="500px" src="/static/blog/cleanrl-v1/rlops.png">
+
```

Comment:
Do you plan on merging RLops before 1.0 release?
This is coming off as it is released in 1.0
Maybe a little rewording would be good if that is not the case.

Reply:
Do not plan to merge it before 1.0 release. I changed the phrasing to “we are building an RLops utility”